src/common/storage/socketio/websocket.js

```diff
@@ -174,6 +174,19 @@ define([
         };
 
         this.makeCommit = function (data, callback) {
+            var localCoreObjects = data.coreObjects,
+                i,
+                keys = Object.keys(data.coreObjects || {});
+
+            if (gmeConfig.storage.patchRootCommunicationEnabled && data.patchRoot) {
+                data.coreObjects = {};
+                for (i = 0; i < keys.length; i += 1) {
+                    data.coreObjects[keys[i]] = localCoreObjects[keys[i]];
+                }
+                data.coreObjects[data.commitObject.root] = data.patchRoot;
+            }
+            delete data.patchRoot; //we remove the field in either way
+
```

Comment:
Ohh, you are right. I tought we only have a reference to the coreObjects in the queue.
I will copy/rebuild the whole data then and send that out, so we will still have the reference in the queue